run_tests.sh

```diff
@@ -0,0 +1,277 @@
+#!/bin/bash
+
+specific_test=""
+specific_package=""
+cleanup_id=""
+slow_mode=false
+
+while getopts ":st:p:c:" opt; do
+  case $opt in
+    t) specific_test="$OPTARG" ;;
+    p) specific_package="$OPTARG" ;;
+    c) cleanup_id="$OPTARG" ;;
+    s) slow_mode=true ;;
+    \?) cat <<EOT >&2 && exit 1 ;;
+Invalid option -$OPTARG, valid options are
+  -s to run tests in slow mode (one at a time to avoid AWS rate limiting)
+  -c to run clean up only with the given id (eg. abc123)
+  -t to specify a specific test (eg. TestBase)
+  -p to specify a specific test package (eg. one)
+Only one of -c, -t, or -p can be used at a time.
+EOT
+  esac
+done
+
+if [ $slow_mode == true ]; then
+  echo "Running in slow mode: tests will be run one at a time to avoid AWS rate limiting."
+elif [ $slow_mode == false ]; then
+  echo "Running in normal mode: tests will be run in parallel."
+fi
+if [ -n "$specific_test" ]; then
+  echo "Specific test to run: $specific_test"
+else
+  echo "No specific test to run."
+fi
+if [ -n "$specific_package" ]; then
+  echo "Specific package to run: $specific_package"
+else
+  echo "No specific package to run."
+fi
+if [ -n "$cleanup_id" ]; then
+  echo "Cleanup only mode enabled with id: $cleanup_id"
+fi
+if [ -n "$cleanup_id" ] && { [ -n "$specific_test" ] || [ -n "$specific_package" ]; }; then
+  echo "Error: Only one of -c, -t, or -p can be used at a time." >&2
+  exit 1
+fi
+if [ -n "$specific_test" ] && { [ -n "$specific_package" ] || [ -n "$cleanup_id" ]; }; then
+  echo "Error: Only one of -c, -t, or -p can be used at a time." >&2
+  exit 1
+fi
+if [ -n "$specific_package" ] && { [ -n "$specific_test" ] || [ -n "$cleanup_id" ]; }; then
+  echo "Error: Only one of -c, -t, or -p can be used at a time." >&2
+  exit 1
+fi
+
+# shellcheck disable=SC2143
+if [ -n "$cleanup_id" ]; then
+  export IDENTIFIER="$cleanup_id"
+fi
+
+REPO_ROOT="$(git rev-parse --show-toplevel)"
+
+# Find the tests directory
+TEST_DIR=""
+if [ -d "tests" ]; then
+  TEST_DIR="tests"
+elif [ -d "test/tests" ]; then
+  TEST_DIR="test/tests"
+elif [ -d "test" ]; then
+  TEST_DIR="test"
+else
+  echo "Error: Unable to find tests directory" >&2
+  exit 1
+fi
+
+run_tests() {
+  local slow_mode=$1
+  local TEST_DIR="$2"
```

Comment:
use lower case for local variable 
```suggestion
  local test_dir="$2"
```